packages/hash/api/src/graphql/resolvers/pages/updatePageContents.ts

```diff
@@ -74,17 +76,74 @@ export const updatePageContents: Resolver<
         }),
     );
 
+    // Perform any block data swapping updates.
+    await Promise.all(
+      actions
+        .map(({ swapBlockData }) => swapBlockData)
+        .filter(
+          (swapBlockData): swapBlockData is SwapBlockData => !!swapBlockData,
+        )
+        .map(async (swapBlockData) => {
+          const block = await Block.getBlockById(client, {
+            accountId: swapBlockData.accountId,
+            entityId: swapBlockData.entityId,
+          });
+
+          const blockDataEntity = await Entity.getEntityLatestVersion(client, {
+            accountId: swapBlockData.newEntityAccountId,
+            entityId: swapBlockData.newEntityEntityId,
+          });
+
+          if (!block) {
+            throw new Error(
+              `Block with entityId ${swapBlockData.entityId} not found`,
+            );
+          }
+
+          if (!blockDataEntity) {
+            throw new Error(
+              `Target entity with entityId ${swapBlockData.newEntityEntityId} not found`,
+            );
+          }
+
+          const blockEntity = await Entity.getEntityLatestVersion(client, {
+            accountId: swapBlockData.accountId,
+            entityId: swapBlockData.entityId,
+          });
+
+          // forced updatedAt property to update.
+          // This is done because the frontend currently uses to `updatedAt` property
+          // to determine the most recent entity, when it comes across an entity that
+          // is also present in entity store. When a block's data is swapped, the updatedAt
+          // field of the block doesn't change and hence the current frontend setup won't know
+          // the most recent entity.
+          //
+          // This shouldn't be done since we can check if a block's data has been swapped by
+          // checking `createdAt` value of the outgoing `$.data` link.
+          // This is should be removed once the frontend has been refactored to full rely on links
+          await blockEntity?.updateEntityProperties(client, {
+            updatedByAccountId: swapBlockData.accountId,
+            properties: blockEntity.properties,
+          });
```

Comment:
As explained in the comment, in the current setup, we can't know if a block's data has been swapped, since the `updatedAt` property is not updated in that scenario. This temporarily updates that property so that blocks don't get reset when we swap a block's data. We might not need to do this if we update the frontend setup to fully rely on links.